Cross-version initialization

I noticed something odd, as I was plucking some libraries together for a small rendering test.

Apparently, OpenGL is not initialized cross-version. What this effectively means is that e.g. a main program which initializes OpenGL 3.3 and a library that uses (or rather - assumes) OpenGL 3.2, will not work together. See the attached code for an fail-fast example:

```go
package main

import (
	gl32 "github.com/go-gl/gl/v3.2-core/gl"
	_ "github.com/go-gl/gl/v3.3-core/gl"
	"github.com/go-gl/glfw/v3.2/glfw"
)

func init() {
	runtime.LockOSThread()
}

func main() {
	// Ensure valid context
	if err := glfw.Init(); err != nil {
		closer.Fatalln(err)
	}
	win, err := glfw.CreateWindow(800, 600, "Demo", nil, nil)
	if err != nil {
		closer.Fatalln(err)
	}
	win.MakeContextCurrent()

	if err := gl32.Init(); err != nil {
		closer.Fatalln("opengl: init failed:", err)
	}
}
```

This will fail to compile under Ubuntu 16.04 with the following message:
```
# command-line-arguments
/usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/tmp/go-link-284705261/000001.o: In function `GlowGetProcAddress':
/home/bgerda/go/src/github.com/go-gl/gl/v3.3-core/gl/procaddr.go:54: multiple definition of `GlowGetProcAddress'
/tmp/go-link-284705261/000000.o:/home/bgerda/go/src/github.com/go-gl/gl/v3.2-core/gl/procaddr.go:54: first defined here
collect2: error: ld returned 1 exit status
```

I think I remember that my code did compile under Windows 10, but failed at runtime (I might mistaken though - will check later today. Something about multiple definitions, too).

I was under the assumption that this should work, as right now it is impossible to combine libraries which use different versions of go-gl ([nuklear](https://github.com/golang-ui/nuklear) e.g. uses 3.2 - which clashes with my code which is based upon 3.3).

--- a/all-core/gl/procaddr.go
+++ b/all-core/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcoreall(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcoreall(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcoreall(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcoreall(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcoreall(cname)
 }

--- a/v2.1/gl/procaddr.go
+++ b/v2.1/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_gl21(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_gl21(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_gl21(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_gl21(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_gl21(cname)
 }

--- a/v3.1/gles2/procaddr.go
+++ b/v3.1/gles2/procaddr.go
@@ -23,7 +23,7 @@ package gles2
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_gles231(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gles2
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_gles231(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gles2
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_gles231(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_gles231(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_gles231(cname)
 }

--- a/v3.2-compatibility/gl/procaddr.go
+++ b/v3.2-compatibility/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility32(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility32(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility32(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility32(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcompatibility32(cname)
 }

--- a/v3.2-core/gl/procaddr.go
+++ b/v3.2-core/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore32(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore32(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore32(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore32(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcore32(cname)
 }

--- a/v3.3-compatibility/gl/procaddr.go
+++ b/v3.3-compatibility/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility33(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility33(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility33(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility33(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcompatibility33(cname)
 }

--- a/v3.3-core/gl/procaddr.go
+++ b/v3.3-core/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore33(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore33(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore33(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore33(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcore33(cname)
 }

--- a/v4.1-compatibility/gl/procaddr.go
+++ b/v4.1-compatibility/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility41(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility41(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility41(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility41(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcompatibility41(cname)
 }

--- a/v4.1-core/gl/procaddr.go
+++ b/v4.1-core/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore41(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore41(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore41(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore41(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcore41(cname)
 }

--- a/v4.2-compatibility/gl/procaddr.go
+++ b/v4.2-compatibility/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility42(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility42(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility42(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility42(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcompatibility42(cname)
 }

--- a/v4.2-core/gl/procaddr.go
+++ b/v4.2-core/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore42(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore42(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore42(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore42(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcore42(cname)
 }

--- a/v4.3-compatibility/gl/procaddr.go
+++ b/v4.3-compatibility/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility43(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility43(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility43(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility43(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcompatibility43(cname)
 }

--- a/v4.3-core/gl/procaddr.go
+++ b/v4.3-core/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore43(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore43(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore43(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore43(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcore43(cname)
 }

--- a/v4.4-compatibility/gl/procaddr.go
+++ b/v4.4-compatibility/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility44(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility44(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility44(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility44(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcompatibility44(cname)
 }

--- a/v4.4-core/gl/procaddr.go
+++ b/v4.4-core/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore44(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore44(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore44(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore44(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcore44(cname)
 }

--- a/v4.5-compatibility/gl/procaddr.go
+++ b/v4.5-compatibility/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility45(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility45(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility45(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcompatibility45(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcompatibility45(cname)
 }

--- a/v4.5-core/gl/procaddr.go
+++ b/v4.5-core/gl/procaddr.go
@@ -23,7 +23,7 @@ package gl
 #if defined(TAG_EGL)
 	#include <stdlib.h>
 	#include <EGL/egl.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore45(const char* name) {
 		return eglGetProcAddress(name);
 	}
 #elif defined(TAG_WINDOWS)
@@ -31,7 +31,7 @@ package gl
 	#include <windows.h>
 	#include <stdlib.h>
 	static HMODULE ogl32dll = NULL;
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore45(const char* name) {
 		void* pf = wglGetProcAddress((LPCSTR) name);
 		if (pf) {
 			return pf;
@@ -44,13 +44,13 @@ package gl
 #elif defined(TAG_DARWIN)
 	#include <stdlib.h>
 	#include <dlfcn.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore45(const char* name) {
 		return dlsym(RTLD_DEFAULT, name);
 	}
 #elif defined(TAG_POSIX)
 	#include <stdlib.h>
 	#include <GL/glx.h>
-	void* GlowGetProcAddress(const char* name) {
+	void* GlowGetProcAddress_glcore45(const char* name) {
 		return glXGetProcAddress(name);
 	}
 #endif
@@ -61,5 +61,5 @@ import "unsafe"
 func getProcAddress(namea string) unsafe.Pointer {
 	cname := C.CString(namea)
 	defer C.free(unsafe.Pointer(cname))
-	return C.GlowGetProcAddress(cname)
+	return C.GlowGetProcAddress_glcore45(cname)
 }

